### ISSUE TITLE
Typescript extensions should include tsx

### ISSUE THREAD

**Issue by heygrady** [eslint-community/eslint-plugin-n#47]
This plugin supports Typescript's Node ESM file extension conventions but does not support all Typescript extensions.

In this case, `Example.tsx` exists but is not properly resolved.

```
error  "./Example.js" is not found  n/no-missing-import
```

The Typescript docs mention the following extensions:
- ts
- tsx <-- official typescript file extension
- cts
- mts

https://github.com/weiran-zsd/eslint-plugin-node/blob/d7b975a07e1b876ca2c75597d054a81564b25685/lib/util/is-typescript.js#L5

https://github.com/weiran-zsd/eslint-plugin-node/blob/d7b975a07e1b876ca2c75597d054a81564b25685/lib/util/map-typescript-extension.js#L6-L11

This would probably fix it:

```js
// lib/util/is-typescript.js
const typescriptExtensions = [".ts", ".tsx", ".cts", ".mts"]
```

```js
// lib/util/map-typescript-extension.js
const mapping = {
    "": ".js", // default empty extension will map to js
    ".ts": ".js",
    ".tsx": ".js", // <--
    ".cts": ".cjs",
    ".mts": ".mjs",
}
```

**Comment by aladdin-add** [comment]
PRs are welcome! 👍

**Comment by scagood** [comment]
This should be resolved now due to #108

**Comment by scagood** [comment]
I am going to assume that this is resolved, and thus I will close this issue.

Please re open it if you still have this issue!
